docs/oauth-setup.md

```diff
@@ -0,0 +1,143 @@
+# Using an external OAuth provider with S.H.I.E.L.D
+
+## Github
+
+### Open the Github settings page
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github1.png "github OAuth")
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github2.png "github OAuth")
+
+### Open "OAuth Apps" under the "Developer settings" field on the lower left menu
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github3.png "github OAuth")
+
+### Register a new Github OAuth client with the "Register a new application" button
+
+![Github Config Image](https://raw.githubusercontent.com/starkandwayne/shield/oauth-docs/docs/images/githubAuth/github4.png "github OAuth")
+
+- The only field that is SHIELD specific here is the "Authorization callback URL" field, which needs to be:
+  - `http://<YOUR_SHIELD_ADDRESSS_HERE>/auth/oauth/<YOUR_GH_IDENTIFIER_FROM_YOUR_SHIELD_CONFIG_HERE>/redir`
```

Comment:
Can we make these placeholders smaller?  I.e. `$shield` and `$identifier`, and just call out in the surrounding prose what `$shield` and `$identifier` mean?